Include/internal/pycore_typeobject.h

```diff
@@ -89,6 +89,9 @@ _PyType_GetModuleState(PyTypeObject *type)
 // function
 PyAPI_FUNC(PyObject *) _PyType_GetDict(PyTypeObject *);
 
+PyAPI_FUNC(PyObject *) _PyType_LookupSubclasses(PyTypeObject *);
+PyAPI_FUNC(PyObject *) _PyType_InitSubclasses(PyTypeObject *);
```

Comment:
It feels a little off, putting the dict out there to be modified. What do you need this for, specifically?  Can we provide some specific operation (like "set a dict watcher") rather than exposing the whole dict?

Reply:
I'll drop this for now and come back and do a different PR that specifically addresses the classes. It seems like adding a whole new API for this is deserving of its own PR.